src/views/PanelViews/DepartmentViews/DepartmentPageView/DepartmentPageView.js

```diff
@@ -0,0 +1,59 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useNavigate, useOutlet, useParams } from 'react-router-dom';
+import { SessionContext } from '../../../../contexts';
+import { roles } from '../../../../shared';
+
+export function DepartmentPageView() {
+    const { departmentID } = useParams();
+    const { userData } = useContext(SessionContext);
+    const [departmentData, setDepartmentData] = useState();
+    const navigate = useNavigate();
+    const outlet = useOutlet();
+
+    const userRole = userData.role;
```

Comment:
When you'll have time, learn what `useMemo` and `useCallback` hooks do. For now, it is enough, that this just works